### PR TITLE
[MetaSearch] support OGC API - Records in default connections (#46336)

### PR DESCRIPTION
**Pull request by tomkralidis**
## Description

As per #46336, this PR adds OGC API - Records support for MetaSearch default connections.

~~Note that this PR includes a breaking change to file format of https://github.com/qgis/QGIS/blob/master/python/plugins/MetaSearch/resources/connections-default.xml.  Trying to load the previous format will now yield an error with ("Invalid XML format").  If this is too risky, I'm happy to discuss workarounds/alternatives.~~

The documentation will also need updating as a result.

cc  @bo-lu

### CODE REVIEW

**Comment by nyalldawson** [comment]
>  Trying to load the previous format will now yield an error with ("Invalid XML format"). 

Can you explain further why this is necessary? As far as I can see we've just got the new "type" attribute added?

**Comment by tomkralidis** [comment]
> > Trying to load the previous format will now yield an error with ("Invalid XML format").
> 
> Can you explain further why this is necessary? As far as I can see we've just got the new "type" attribute added?

Correct, we are extending the MetaSearch default connections list with a `type` attribute to help with detecting OGC:CSW or OARec, which was implemented in #44812.

**Comment by nyalldawson** [comment]
> Correct, we are extending the MetaSearch default connections list with a type attribute to help with detecting OGC:CSW or OARec, which was implemented in #44812.

Couldn't we leave the element names as qgsCSWConnections / catalog, and then just check if the type attribute is present to switch to the newer behaviour? Then we'd still be compatible with older files...

**Comment by tomkralidis** [comment]
> > Correct, we are extending the MetaSearch default connections list with a type attribute to help with detecting OGC:CSW or OARec, which was implemented in #44812.
> 
> Couldn't we leave the element names as qgsCSWConnections / catalog, and then just check if the type attribute is present to switch to the newer behaviour? Then we'd still be compatible with older files...

Done.  PR updated.

File: python/plugins/MetaSearch/dialogs/manageconnectionsdialog.py [hunk 1]
```diff
@@ -164,7 +166,7 @@ def load(self, items):
 
             # no dups detected or overwrite is allowed
             url = '/MetaSearch/%s/url' % conn_name
-            self.settings.setValue(url, csw.attrib.get('url'))
+            self.settings.setValue(url, catalog.attrib.get('url'))
```
Comment (nyalldawson) [inline python/plugins/MetaSearch/dialogs/manageconnectionsdialog.py:169]: shouldn't this be setting the type value here too?

Comment (tomkralidis) [reply python/plugins/MetaSearch/dialogs/manageconnectionsdialog.py:169]: fixed/pushed.

**Review by nyalldawson — APPROVED** [review]

**Pull request merged**